metrics/lastseen/tracker.go

```diff
@@ -0,0 +1,226 @@
+package lastseen
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/wrapp/instrumentation/logs"
+)
+
+// Tracker takes care of tracking and exporting metrics
+type Tracker interface {
+	// SetSeen records a seen timestamp for the given field
+	SetSeen(ctx context.Context, field string)
+	// GetSeen returns the unix time stamp when the given field was last tracked as seen
+	GetSeen(ctx context.Context, field string) int64
+	// Wait for all exports to finish
+	WaitForExporters()
+}
+
+// LastSeen keeps track of a last seen and its last flush time
+type LastSeen struct {
+	Value     int64
+	LastFlush int64
+}
+
+// TrackerOption defines a configuration function
+type TrackerOption func(*TrackerOptions)
```

Comment:
What  do you think about renaming `TrackerOption` to  `TrackerFunc` : I found  it confusing at first to have `TrackerOption` and  `TrackerOptions` :)

Reply:
Yeah, a bit confusing

Reply:
Changed to TrackerFunc